Restore the 3.x behavior of an empty -I '' argument.

Bug: http://bugs.marples.name/show_bug.cgi?id=116
Patch: forward-ported from 4.0.1 version.

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar --exclude '*.orig' --exclude '*.rej' dhcpcd-4.99.2.orig/if-options.c dhcpcd-4.99.2/if-options.c
--- dhcpcd-4.99.2.orig/if-options.c	2008-09-19 06:36:40.000000000 -0700
+++ dhcpcd-4.99.2/if-options.c	2008-09-25 01:27:46.115364821 -0700
@@ -494,7 +494,7 @@
 	case 'I':
 		/* Strings have a type of 0 */;
 		ifo->clientid[1] = 0;
-		if (arg)
+ 		if (arg && *arg)
 			s = parse_string_hwaddr((char *)ifo->clientid + 1,
 						CLIENTID_MAX_LEN, arg, 1);
 		else
